File: src/main/java/client/messages/broadcast/types/AlertMessage.java
package client.messages.broadcast.types;

import client.messages.broadcast.AbstractBroadcastMessage;
import client.messages.broadcast.BroadcastMessageType;
import util.packet.PacketWriter;

/**
 * Color:
 * Prefix:
 * <p>
 * Alert popup
 */
public class AlertMessage extends AbstractBroadcastMessage {

    private final String message;

    public AlertMessage(String message) {
        this.message = message;
    }

    @Override
    public BroadcastMessageType getType() {
        return BroadcastMessageType.ALERT;
    }

    @Override
    protected void encodeData(PacketWriter pw) {
        pw.writeMapleString(message);
    }
}

File: src/main/java/client/messages/broadcast/types/AlertMessage.kt
package client.messages.broadcast.types

import client.messages.broadcast.AbstractBroadcastMessage
import client.messages.broadcast.BroadcastMessageType
import util.packet.PacketWriter

/**
 * Color:
 * Prefix:
 *
 * Alert popup
 */
class AlertMessage(private val message: String) : AbstractBroadcastMessage() {

    override val type: BroadcastMessageType get() = BroadcastMessageType.ALERT

    override fun encodeData(pw: PacketWriter) {
        pw.writeMapleString(message)
    }
}